LLPLabelProducer/plugins/LLPGenDecayInfoProducer.cc

```diff
@@ -149,6 +149,16 @@ LLPGenDecayInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSet
                             {
                                 throw cms::Exception("GenParticle relations not properly setup!");
                             }
+
+                            // Handle taus
+                            if (abs(daughter->pdgId) == 15){
+                                // check for leptonic decays
+                                for(size_t i = 0; i < daughter->numberOfDaughters(); ++i) {
+                                    if (abs(daughter->daughter(i)->pdgId()) == 11 or abs(daughter->daughter(i)->pdgId()) == 13){
+                                        daughter = daughter->daughter(i);
```

Comment:
not sure if mu/e are direct daughters of taus or if you have to also account for cases when there is an intermediate W boson stored in the event

Reply:
I think the W* is cleaned in the gen particle collection we're using